interview-questions.md

```diff
@@ -8,36 +8,59 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: The RESTful routes must always need params are show, edit, update, amd destroy. With create, it requires strong params to ensure that the database stores only the desired data. The controlloer will be able to figure out which id to fetch on to create data when the params are then passed. 
 
-Researched answer:
+Researched answer: Additionally, all routes require an id in order for them to run. 
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer: Model, migration, and controller.
+- Model: Allows user to create a new table in the database that includes columns and the data type they want.
+-Migration: It will add any columns that were created in the previous model.
+-Controller: Allow user to create or view database
 
-Researched answer:
+Researched answer: Model is also knwon as CRUD. Model allows user to add, update or remove data from a database, similar to using excel. Migration allows the user to alter the schema and database independently without having to manually enter. Controlloer adds methods to a file that was generated and can be defined the outcome of what the user wants when the method is specified. 
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
-action: "GET" location: /students
+action: "GET" location: /students 
+    -  Display all instances of students and is linked to the index method
 
 action: "POST" location: /students
+    - Allows user to create new students in the database and is linked to the create method
 
 action: "GET" location: /students/new
+    - Allows user to view some form to create a new student and is linked to the new method
 
 action: "GET" location: /students/2
+    - Displays the student with the id of 2 and is linked to the show method
 
 action: "GET" location: /students/2/edit
+    - Displays a form to edit the student with the id of 2 and is linked to the edit method
 
 action: "PATCH" location: /students/2
+    - Allows the user to directly cahnge any values of the student with id 2 and is linked to the update method
 
 action: "DELETE" location: /students/2
+    - Deletes the entire instants of student id 2 and is linked to the destroy method.
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+    - As a dev, I want to create a to do list and save it in the database.
```

Comment:
One thing to note is the difference between a developer story and a user story.  User stories are going to discuss a specific feature that a user can experience in an application versus a developer story that will be more technical.  Project managers will often assign user stories to a developer, and as a developer, you can break down the steps to achieve that goal into developer stories.